### ISSUE TITLE
Add identity column support to XMLGenerator

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#6141]
This is currently not supported yet:
https://github.com/jOOQ/jOOQ/blob/version-3.9.2/jOOQ-codegen/src/main/java/org/jooq/util/XMLGenerator.java#L125

More info here:
http://stackoverflow.com/questions/43552078/jooq-exclude-id-field-when-inserting-from-pojo/43557620?noredirect=1#comment74198963_43557620

**Comment by bisoldi** [comment]
@lukaseder I got part of the way towards a work-around.  I followed your two work-around suggestions on SO (set syntheticIdentity as well as reset) and I had to actually use both.  So, I added the following into the pom within the code generator section:

`<syntheticIdentities>^.*\..*\.id$</syntheticIdentities>`

I then also had to call `newRecord.reset("id")` AND `newRecord.reset("created_date")` in order to allow Postgres to generate both the id and the created_date fields.

The problem then was, when calling `T into = newRecord.into(record);` and even `newRecord.get("id")`, both return null.  Additionally, the created_date field in the object (createdDate) also does not show up in Lombok's @toString either.  Very strange.

So, we went back to generating the objects directly from the database (which does work - mostly) and I ran a diff on the two sets of jOOQ table objects (generated from the XML vs. generated from the DB) and the only real difference is the inclusion of INTEGER.nullable(false).

Generated via XML

`public final TableField<SocialNetworkRecord, Integer> ID = createField("id", org.jooq.impl.SQLDataType.INTEGER.defaultValue(org.jooq.impl.DSL.field("nextval('social_network_id_seq'::regclass)", org.jooq.impl.SQLDataType.INTEGER)), this, "");`


Generated via DB

`public final TableField<SocialNetworkRecord, Integer> ID = createField("id", org.jooq.impl.SQLDataType.INTEGER.nullable(false).defaultValue(org.jooq.impl.DSL.field("nextval('social_network_id_seq'::regclass)", org.jooq.impl.SQLDataType.INTEGER)), this, "");`

So, it looks like the prohibition of null value for the `id` (identity) column is what fixes it.

Finally, for some reason, with the objects created directly from the database, jOOQ doesn't recognize that columns that are autogenerated by PostgreSQL (e.g. a timestamp column with `now()` being the PostgreSQL default value) don't need to be inserted and I have to `.reset("created_date")` on the column in order for jOOQ to realize it shouldn't be set.  It's almost as if jOOQ thinks the value has been changed, even though it's a brand new object and the created_date field is null.  Is there a way around having to do this?

This is labeled as fixed, so I thought I would follow up, has it been fixed and still being planned for 3.10.0?  If there is any further info I can provide, let me know.

Thank you!

**Comment by lukaseder** [comment]
> So, we went back to generating the objects directly from the database (which does work - mostly) and I ran a diff on the two sets of jOOQ table objects (generated from the XML vs. generated from the DB) and the only real difference is the inclusion of INTEGER.nullable(false).

Thank you very much for tracking this down. Indeed, this is another bug. I've registered a separate issue for that: https://github.com/jOOQ/jOOQ/issues/6163

> So, it looks like the prohibition of null value for the id (identity) column is what fixes it.

Indeed. This is what I meant on Stack Overflow by *"The only exception to the above behaviour is when the identity column is NOT NULL and the Record value for the identity is null and jOOQ's meta model is aware of both:"*

> It's almost as if jOOQ thinks the value has been changed, even though it's a brand new object and the created_date field is null.

Yes, the semantics of the `Record.changed()` flag is that a changed value must be included in all `INSERT` *and* `UPDATE` statements.

> Is there a way around having to do this?

It's always the same: jOOQ must be aware of a `DEFAULT` expression *and* of non-nullability, in case of which the Java `null` value can only mean: "apply the database `DEFAULT`". Otherwise, jOOQ wouldn't know if you actually want to insert the Java `null` value, so it cannot make any assumptions.

> This is labeled as fixed, so I thought I would follow up, has it been fixed and still being planned for 3.10.0?

The issue described in the header is fixed for 3.10 and will be merged to 3.9.3 (https://github.com/jOOQ/jOOQ/issues/6142).

**Comment by lukaseder** [comment]
I discovered another issue: `UNIQUE` keys are not generated correctly: https://github.com/jOOQ/jOOQ/issues/6165 (they may override the primary key information).

I've noticed that our integration tests only covered importing an manually written XML file using an `XMLDatabase`. I've added a new integration test that also checks if the direct code generation from the database produces the same result as the one from DB -> XML -> Java code.
